src/Microsoft.AI.Agents.Dapr/Agent.cs

```diff
@@ -1,6 +1,27 @@
-﻿namespace Microsoft.AI.Agents.Dapr;
+﻿using Dapr.Actors.Runtime;
+using Dapr.Client;
+using Microsoft.AI.Agents.Abstractions;
 
-public class Agent
+namespace Microsoft.AI.Agents.Dapr;
+
+public abstract class Agent : Actor, IAgent
 {
+    private readonly DaprClient daprClient;
+
+    protected Agent(ActorHost host, DaprClient daprClient) : base(host)
+    {
+        this.daprClient = daprClient;
+    }
+    public abstract Task HandleEvent(Event item);
 
+    public async Task PublishEvent(string ns, string id, Event item)
+    {
+        var metadata = new Dictionary<string, string>() {
+                 { "cloudevent.Type", item.Type },
+                 { "cloudevent.Subject",  item.Subject },
+                 { "cloudevent.id", Guid.NewGuid().ToString()}

```

Comment:
why is cloudevent.id different than id?

Reply:
The cloudevent.id is the metadata property of the CloudEvent envelope created by Dapr, while the cloudevent.Subject + Type are used for Dapr to figure out where to route the event.